packages/flame/lib/src/game/game.dart

```diff
@@ -190,7 +190,10 @@ abstract mixin class Game {
       );
     }
     _gameRenderBox = gameRenderBox;
-    onAttach();
+    if (!isInternalRefresh) {
+      onAttach();
+    }
+    isInternalRefresh = false;
```

Comment:
How come this is set to false here? 🤔

Reply:
The default state is false.  The order is detach then attach.  If you set it to false after callback in the `refreshWidget` or in `detach` where it originally was, then attach will still get called because it would be true by then.  No matter what, attach is always going to be called so its putting it back to its initial state.

Reply:
Right, that makes sense, I hope there won't be any events in-between detach and attach that could be affected in the future.